tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc

```diff
@@ -1797,13 +1797,39 @@ Status TileTransposer::TransposeNode(TransposeContext* context,
 Status UnaryGradTransposer::TransposeNode(TransposeContext* context,
                                           utils::MutableNodeView* node) {
   DCHECK(IsUnaryGrad(*node->node()));
-  if (!ShouldProcess(*context, *node) || !IsFanoutPortRankN(*node, 0, 4) ||
+  const auto* output_shape_attr = node->GetAttr(kAttrOutputShape);
+  const auto& shape = output_shape_attr->list().shape(0);
+  const int rank = shape.dim_size();
+  std::string src_format = context->src_format;
```

Comment:
Can `ScopedDataFormatUpgrader` (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc#L173) be used instead? https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc#L719-L736 is an example of it being used.